### PR TITLE
copr: support insertutf8 and lowerutf8

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What is changed and how it works?

Issue Number: Ref https://github.com/tikv/tikv/issues/5751

What's Changed:

String sig:

* insertutf8 / lowerutf8

```commit-message
```

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/tidb_query_expr/src/impl_string.rs [hunk 1]
```diff
@@ -502,6 +502,35 @@ pub fn insert(
...
+        return Ok(writer.write_ref(Some(s_utf8)));
+    }
+    if ulen > slen - upos + 1 || len < 0 {
+        ulen = slen - upos + 1;
+    }
+    let mut ret = String::with_capacity(newstr.chars().count() + slen);
+    ret.push_str(&s[0..upos - 1]);
+    ret.push_str(newstr);
+    ret.push_str(&s[upos + ulen - 1..]);
+    Ok(writer.write_ref(Some(ret.as_bytes())))
```
Comment (skyzh) [inline components/tidb_query_expr/src/impl_string.rs:531]: We can use partial_write API to avoid allocating a new String.

**Review by skyzh — COMMENTED** [review]
Rest LGTM, thanks!

**Review by skyzh — APPROVED** [review]

**Review by longfangsong — APPROVED** [review]

**Pull request merged**